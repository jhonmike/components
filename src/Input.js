import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'

// import styles from './input.css'

const Input = props => {
  const {
    className,
    type,
    value,
    ...otherProps
  } = props

  const classes = classnames(
    // styles.input,
    className
  )

  return (
    <input {...otherProps}
      type={type}
      className={classes}
      value={value}
    />
  )
}

Input.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string
}

export default Input
