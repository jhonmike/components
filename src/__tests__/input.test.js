import React from 'react'
import renderer from 'react-test-renderer'
import Input from '../Input'

describe('Input Component', () => {
  it('renders text component', () => {
    const tree = renderer.create(<Input type="text" value={''} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders email component', () => {
    const tree = renderer.create(<Input type="email" value={''} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders checkbox component', () => {
    const tree = renderer.create(<Input type="checkbox" checkbox={false} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders radio component', () => {
    const tree = renderer.create(<Input type="radio" name="gender" value="male" checked={true} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
