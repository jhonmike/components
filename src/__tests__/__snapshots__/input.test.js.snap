// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders checkbox component 1`] = `
<input
  checkbox={false}
  className=""
  type="checkbox"
/>
`;

exports[`Input Component renders email component 1`] = `
<input
  className=""
  type="email"
  value=""
/>
`;

exports[`Input Component renders radio component 1`] = `
<input
  checked={true}
  className=""
  name="gender"
  type="radio"
  value="male"
/>
`;

exports[`Input Component renders text component 1`] = `
<input
  className=""
  type="text"
  value=""
/>
`;
